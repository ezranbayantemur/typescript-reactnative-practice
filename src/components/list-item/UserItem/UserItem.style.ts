import {StyleSheet} from 'react-native';
import Colors from '../../../styles/colors';
import Spacing from '../../../styles/spacing';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.smokeGray,
    padding: Spacing.normal,
    margin: Spacing.normal,
    borderRadius: 5,
    flexDirection: 'row',
    alignItems: 'center',
  },
  avatar: {
    width: 50,
    height: 50,
    borderRadius: 25,
  },
  username: {
    fontSize: 20,
    fontWeight: 'bold',
  },
});
