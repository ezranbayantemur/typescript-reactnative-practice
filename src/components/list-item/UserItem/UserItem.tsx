import React, {FC} from 'react';
import {View, Text, Image} from 'react-native';
import {User} from '../../../types';
import styles from './UserItem.style';

export interface UserItemProps {
  user: Partial<User>;
}

const UserItem: FC<UserItemProps> = ({user}) => {
  return (
    <View testID="user-item" style={styles.container}>
      <Image style={styles.avatar} source={{uri: user.avatar}} />
      <View>
        <Text style={styles.username}>{user.username}</Text>
        <Text>
          {user.first_name} {user.last_name}
        </Text>
      </View>
    </View>
  );
};

export default UserItem;
