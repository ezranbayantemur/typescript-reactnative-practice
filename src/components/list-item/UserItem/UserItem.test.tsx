import React from 'react';
import {render} from '@testing-library/react-native';
import UserItem from './UserItem';

test('should render correctly', () => {
  const comp = render(<UserItem user={{}} />);
  expect(comp).toMatchSnapshot();
});
