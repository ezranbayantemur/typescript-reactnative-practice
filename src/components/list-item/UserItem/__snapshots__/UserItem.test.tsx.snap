// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eceff1",
      "borderRadius": 5,
      "flexDirection": "row",
      "margin": 10,
      "padding": 10,
    }
  }
  testID="user-item"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text>
       
    </Text>
  </View>
</View>
`;
